pkgs/development/compilers/ghc/8.0.1.nix

```diff
@@ -35,6 +37,7 @@ stdenv.mkDerivation rec {
     "--with-gcc=${stdenv.cc}/bin/cc"
     "--with-gmp-includes=${gmp.dev}/include" "--with-gmp-libraries=${gmp.out}/lib"
     "--with-curses-includes=${ncurses.dev}/include" "--with-curses-libraries=${ncurses.out}/lib"
+    "--datadir=$doc/share/doc/ghc"
```

Comment:
According to [GHC wiki](https://ghc.haskell.org/trac/ghc/wiki/Building/Installing#Layoutoftheinstalledfiles), `docdir` defaults to `$(prefix)/share/doc/ghc` and can be changed with `--datadir` option (I find this odd, too).